pilot/pkg/model/jwks_resolver.go

```diff
@@ -340,21 +339,27 @@ func (r *JwksResolver) refresher() {
 	for {
 		select {
 		case <-r.refreshTicker.C:
-			r.refresh()
+			refreshInterval := r.refresh()
+			// update refresh interval on change.
+			if r.refreshInterval != refreshInterval {
+				r.refreshTicker.Stop()
+				r.refreshTicker = time.NewTicker(refreshInterval)
```

Comment:
This means if the jwks server is not working for some reason, we will keep sending the request every 1 minutes until it works, will this cause too much traffic to the server or puts too much burden on istiod? Not sure if we need to increase the interval after some failed attempt. @incfly  @howardjohn wdyt?

Reply:
Otherwise we could also use an exponential backoff retry mechanism. Starting with 1 minute and then increase it until it reaches the default interval in case it keeps failing.